### ISSUE TITLE
MapAveragingCallback schedules all cylinders at once

### ISSUE THREAD

**Issue by mck1117** [rusefi/rusefi#974]
![image](https://user-images.githubusercontent.com/568254/66712464-ced96b80-ed51-11e9-813f-e7ffab2fc7f5.png)

Scheduling (2 * N) callbacks at once seems like a not great idea.  `MapAveragingTriggerCallback` alone took 26us on a release build, but only one per revolution.  This has significant potential to cause jitter.

**Comment by mck1117** [comment]
And a corollary: should there be an API for scheduling a batch of events, without calling `DoExecute` per item (only call it once at the end)?

![image](https://user-images.githubusercontent.com/568254/66712485-195ae800-ed52-11e9-824f-28b7959a76e0.png)

(this is is a zoom-in of one of the 16 enqueue events above)

**Comment by mck1117** [comment]
Some napkin math suggests that the current logic on an F4 can't run a v8 over 4000 RPM without missing interrupts.

**Comment by rusefi** [comment]
Great find, thank you!

**Comment by rusefi** [comment]
Nick reported hanging self-stimulation at 6K with 60/2 once he has enabled one of the two tachometer implementations or something like that.

Maybe it was about this issue really?

**Comment by mck1117** [comment]
Another consideration: We can't necessarily amortize the cost across multiple teeth, since some trigger patterns may not have more teeth, forcing our hand.  However, those triggers are also at less risk of having interrupts stomp on one another.

**Comment by rusefillc** [comment]
@mck1117 poke :)

**Comment by rusefillc** [comment]
@FDSoftware this is done, right?

**Comment by FDSoftware** [comment]
> [@FDSoftware](https://github.com/FDSoftware) this is done, right?

yup, was https://github.com/rusefi/rusefi/pull/8099 / https://github.com/rusefi/rusefi/pull/8034
